libafl/src/state/mod.rs

```diff
@@ -532,6 +536,32 @@ where
     }
 }
 
+/// A trait for types that want to expose a stop API
+pub trait Stoppable {
+    /// Check if stop is requested
+    fn stop_requested(&self) -> bool;
+
+    /// Request to stop
+    fn request_stop(&mut self);
+
+    /// Discard the stop request
+    fn discard_stop_request(&mut self);
```

Comment:
`cancel_stop_request`? Also, is it really needed?

Reply:
if we are saving the state for restarts, then yes

Reply:
fair

Reply:
I would keep discard in that case

Reply:
We could also cancel it before it even triggers

Reply:
which is the same as discarding it :)

Reply:
marking as resolved:)